### DISCUSSION TITLE
Features

### DISCUSSION THREAD

**Discussion by amalizzio** [disentcorp/dateroll#6] [Ideas]
@Batu4  do you you have the feature list for dateroll? if you can paste it here, we can add them one by one into issues as enhancement requests

**Comment by Batu4** [comment]
roadmap for dateroll
1 new release per week
bugfixes and at least 1 new feature

date adjusters
5/5/25|FCY    frst day curr yr
5/5/25|FPY    frst day prev yr
5/5/25|FNY  frst day next yr
5/5/25|LCY    last day curr yr
5/5/25|LPY    last day prev yr
5/5/25|LNY    last day next yr
5/5/25|FCY    frst day curr yr
5/5/25|FPM    frst day prev mth
5/5/25|FNM    frst day next mth
5/5/25|LCM    last day curr mth
5/5/25|LPM    last day prev mth
5/5/25|LNM    last day next mth
5/5/25|FPM    frst day prev mth
5/5/25|FNQ    frst day next qtr
5/5/25|LCQ    last day curf qtr
5/5/25|LPQ    last day prev qtr
5/5/25|LNQ    last day next qtr
5/5/25|1WCM    1st  wed curr mth
5/5/25|2FCM 2nd fri curr mth
^ prob more, week cases

call ddh on a list of thigns
if thigns are datelike, parse them
if strlike parse them, yada...

timezone-aware support. 
force utc, add setting to default
needs design

re-architecture
merge calendar and cal math into holiday
merge drawer and safe open

module-level caching with invalidation on calendars, cache every ddh call

add query via date-slicing
outright years, quarters
"2024" -> 1 or more dates, with settings for first and last of set sorted
2024Q2 -> which date? first/last
2024May02 -> which hour? 

schedule add support for stop middle part to be duration appened to first pstart
ddh('t+2y,5y,1m')
stop = t+2y+5y because stop is Duration

add daycounters!!!!!
ACT/365
30/360
everythign on wikipedia!!!!!!!!!!!!!!!!!!
plus document the sources, ISDA, ICMA, ESMA, ...

Schedule
    rule based
    t,t+5y,1m
    1/1/20,1/150,1y
    t-5y|11MCYu3WCMuRCW,+10yr,1y
    t-5y|12MCYu25DCM,+10y,1y
    t-5y|12MCYu25DCMuCBD,+10y,1y

observation rule
CBD = closest business day (sun->mon, sat->fri)
NBD = next business day
PBD = previous business day

add time support
H, M, S, no sub-second
hours first, then minutes, seconds

examples
10 examples per week of something useful

pick a bond/derivative, check static configs in quantlib, strata, and rateslib
pick a countries holidays (rules for mongolian holidays)
lunar calendar conversions from book
solar calendar conversions form book
country-specific implementations
